sycl/test/aot/multiple-devices.cpp

```diff
@@ -0,0 +1,132 @@
+//==----- multiple-devices.cpp - Appropriate AOT-compiled image selection  ----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------------------------===//
+
+// REQUIRES: ioc64, ocloc, aoc
+
+// Produce object file, spirv
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_x86_64-unknown-linux-sycldevice,spir64_gen-unknown-linux-sycldevice,spir64_fpga-unknown-linux-sycldevice %s -c -o %t.o
+// RUN: %clangxx -fsycl -fsycl-link-targets=spir64_x86_64-unknown-linux-sycldevice,spir64_gen-unknown-linux-sycldevice,spir64_fpga-unknown-linux-sycldevice %t.o -o %t.spv
+// AOT-compile device binary images
+// RUN: ioc64 -cmd=build -binary=%t.spv -ir=%t_cpu.ir -device=cpu
+// RUN: ocloc -file %t.spv -spirv_input -output %t_gen.out -output_no_suffix -device cfl
+// RUN: aoc %t.spv -o %t_fpga.aocx -sycl -dep-files=%t.d
+
+// CPU, GPU
```

Comment:
I think you can test spirv fallback on FPGA here instead of separate run.
Same for other devices.

Reply:
This would de facto remove the checks for no SPIR-V image being present in the fat binary. If some change attempts to break the device selection logic, there will be no convenient way to test that an AOT-compiled image can actually be chosen.

Still, I can reduce the `RUN`s if you think this won't lessen the regression coverage.

Reply:
Is it possible to have fat binary without SPIR-V?

Reply:
I've doublechecked with:
`clang++ -fsycl -fsycl-add-targets=spir64_gen:dummy_gen.out dummy.o -ccc-print-phases`
`clang++ -fsycl -fsycl-add-targets=spir64_gen-unknown-linux-sycldevice -Xsycl-target-backend=spir64_gen-unknown-linux-sycldevice "-device skl" dummy.cpp -ccc-print-phases`

In both cases, only the AOT compilation result is linked into the fat binary apart from the host image. SPIR-V is linked either if `spir64-unknown-linux-sycldevice` is specified explicitly or if no target is specified at all (`spir64-unknown-linux-sycldevice` is default then). This seems to make perfect sense.